frontends/p4/toP4/toP4.cpp

```diff
@@ -153,17 +150,17 @@ bool ToP4::preorder(const IR::P4Program *program) {
     dump(2);
     for (auto a : program->objects) {
         // Check where this declaration originates
-        cstring sourceFile = ifSystemFile(a);
-        if (!a->is<IR::Type_Error>() &&  // errors can come from multiple files
-            sourceFile != nullptr) {
+        auto sourceFileOpt = ifSystemFile(a);
+        // Errors can come from multiple files
+        if (!a->is<IR::Type_Error>() && sourceFileOpt.has_value()) {
```

Comment:
As a way of personal preference, I usually prefer casting optional to `bool` to check if there is something inside :)

Reply:
Couldn't this lead to ambiguity if the type of sourceFileOpt changes? Main reason why I always try make this explicit with `has_value`.

Reply:
Only if type would be convertible to `bool`. Note that we need explicit conversion for `std::optional` (and this is better compared to `boost::optional` that is implicitly convertible to bool). Anyway, as I said, this is mostly within personal preference :)

Reply:
Makes sense!